rust/src/jitter.rs

```diff
@@ -0,0 +1,368 @@
+        };
+
+        let mut auction_subscriber = AuctionSubscriber::new(auction_subscriber_config);
+        auction_subscriber.subscribe().await?;
+
+        let auction_sender = self.auction_sender.clone();
+        auction_subscriber.event_emitter.subscribe("auction",  move |event| {
+            let _ = auction_sender.try_send(event);
+        });
+
+        futures::future::pending::<()>().await;  // Keep the event loop alive forever
+        Ok(())
+    }
+
+    // Simple trampoline to the async function we need from the auction event 
+    pub fn on_auction_sync(self: &Arc<Self>, event: Box<dyn Event>) -> JitResult<()> {
+        let jitter = self.clone();
+        tokio::spawn(async move {
+            let _ = jitter.on_auction(event).await;
+        });
+
+        Ok(())
+    }
+
+    // Process the auction event & attempt to fill with JIT if possible
+    pub async fn on_auction(&self, event: Box<dyn Event>) -> JitResult<()> {
+        if let Some(auction) = event.as_any().downcast_ref::<ProgramAccountUpdate<User>>() {
+            log::info!("Auction received");
+
+            let user_pubkey = auction.pubkey.clone();
+            let user = auction.data_and_slot.data.clone();
+            let slot = auction.data_and_slot.slot;
+            let user_stats_key = Wallet::derive_stats_account(&user.authority, &drift_program);
+
+            for order in user.orders {  
+                if order.status != OrderStatus::Open {
+                    continue;
+                }
+
+                if !order.has_auction_price(order.slot, order.auction_duration, slot)? {
+                    continue;
+                }
+
+                if let Some(user_filter) = &self.user_filter {
+                    if user_filter(&user, &user_pubkey, order.clone()) {
+                        continue;
+                    }
+                }
+
+                let order_sig = self.get_order_signatures(&user_pubkey, order.order_id);
+
+                if let Some(_) = self.ongoing_auctions.get(&order_sig) {
+                    continue;
+                }
+
+                match order.market_type {
+                    MarketType::Perp => {
+                        log::info!("Perp Auction");
+
+                        if let Some(_) = self.perp_params.get(&order.market_index) {
+
+                            let perp_market: PerpMarket = self.drift_client.get_perp_market_info(order.market_index).await?;
+                            
+                            if order.base_asset_amount - order.base_asset_amount_filled <= perp_market.amm.min_order_size {
+                                log::warn!("Order filled within min order size");
+                                log::warn!("Remaining: {}", order.base_asset_amount - order.base_asset_amount_filled);
+                                log::warn!("Minimum order size: {}", perp_market.amm.min_order_size);
+                                return Ok(())
+                            }
+
```

Comment:
is it worth adding another check here that base_asset_remaining will not exceed our min/max pos? would save some sim errors and/or failed txs